File: src/main/java/com/github/jaksonlin/testcraft/application/processors/UnittestMethodProcessor.java
package com.github.jaksonlin.testcraft.application.processors;

import com.github.jaksonlin.testcraft.core.context.UnittestMethodContext;
import com.intellij.psi.PsiComment;
import com.intellij.psi.PsiMethod;
import com.intellij.psi.util.PsiTreeUtil;

import java.util.ArrayList;
import java.util.List;

public class UnittestMethodProcessor {
    public static UnittestMethodContext fromPsiMethod(PsiMethod psiMethod) {
        List<String> comments = extractCommentsFromMethodBody(psiMethod);
        return new UnittestMethodContext(psiMethod.getName(), comments);
    }

    private static List<String> extractCommentsFromMethodBody(PsiMethod psiMethod) {
        List<String> comments = new ArrayList<>();
        for (PsiComment comment : PsiTreeUtil.findChildrenOfType(psiMethod, PsiComment.class)) {
            comments.add(comment.getText());
        }
        return comments;
    }
}

File: src/main/java/com/github/jaksonlin/pitestintellij/processor/UnittestMethodProcessor.kt
package com.github.jaksonlin.pitestintellij.processor

import com.github.jaksonlin.pitestintellij.context.UnittestMethodContext
import com.intellij.psi.PsiComment
import com.intellij.psi.PsiMethod
import com.intellij.psi.util.PsiTreeUtil

class UnittestMethodProcessor {
    companion object {
        fun fromPsiMethod(psiMethod: PsiMethod): UnittestMethodContext {
            val comments = extractCommentsFromMethodBody(psiMethod)
            return UnittestMethodContext(psiMethod.name, comments)
        }

        private fun extractCommentsFromMethodBody(psiMethod: PsiMethod): List<String> {
            val comments = mutableListOf<String>()
            PsiTreeUtil.findChildrenOfType(psiMethod, PsiComment::class.java).forEach {
                comments.add(it.text)
            }
            return comments
        }
    }
}